bin/serverless

```diff
@@ -7,6 +7,20 @@ const BbPromise = require('bluebird');
 const logError = require('../lib/classes/Error').logError;
 const uuid = require('uuid');
 const initializeErrorReporter = require('../lib/utils/sentry').initializeErrorReporter;
+const semver = require('semver');
+const packageJson = require('../package.json');
+
+function checkNodeVersion(serverless) {
+  const requiredVersion = packageJson.engines.node;
+  const nodeVersion = process.version;
+  if (!semver.satisfies(nodeVersion, requiredVersion)) {
+    const msg = [
+      `Serverless Framework requires Node.js version ${requiredVersion}. `,
+      `You're running version ${nodeVersion}.`,
+    ].join('');
+    serverless.cli.log(msg);
```

Comment:
It's a great idea to give a clean notice, still for this to be reliable, it should meet following constraints:
- Should be put before any `require`'s (so new syntax in those modules doesn't crash execution upfront).
- No non supported in Node.js v4 syntax should be used in this file

As I think it'll be hard to enforce above with future updates, I suggest that we introduce a separate _preliminary check_ file as e.g.:

_bin/serverless-with-preliminary-checks.js_
```javascript
'use strict';

// .. logic that confirm Node.js version aborts if it's not met ...

require('./serverless');
```

And point `bin.serverless` in package.json to that one instead.

(_as a side note, I think good convention is to end binary files with `.js` extension. Then they're also out of a box picked by tools as ESLint, Prettier etc._).

Other option is that we skip on this notice now, and just add `enginesStrict: true`.
It'll ensure that npm won't update `serverless` for Node.js v4 users (assuming they also use npm that was bundled with Node.js v4) - new versions will be invisble for them.
That'll work only as far as Node.js v4 is concerned. When we will drop Node.js v6 support, it'll apply only if developers specifically advised npm in its config to do that.

Reply:
Yes, good points. I decided to use `engineStrict` for now. I added another commit which removes the check for the Node version. Introducing different `bin` files might cause some confusion in the long term.

We can still re-visit this later on, when we decide that such a Node.js version check is worth the trouble.